main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"fmt"
+	"math"
```

Comment:
Could you avoid using `math` package here? Using `math.Pow` is just inefficient. You don't need to use any float64 numbers.